### ISSUE TITLE
Crash on invalid JSON file

### ISSUE THREAD

**Issue by marekdedic** [nushell/nushell#1114]
**Describe the bug**
nushell crashes when trying to open an invalid JSON file

**To Reproduce**
```sh
$ echo "{}\n{}" > test.json
$ nu
/home/dir/> open test.json
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: ShellError { error: Diagnostic(ShellDiagnostic { diagnostic: Diagnostic { severity: Error, code: None, message: "Could not parse as JSON", labels: [Label { span: Span { start: 0, end: 4 }, message: Some("input cannot be parsed as JSON"), style: Primary }, Label { span: Span { start: 5, end: 14 }, message: Some("value originates from here"), style: Secondary }] } }), cause: None }', src/libcore/result.rs:1165:5
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace.
$
```

**Expected behavior**
Well, I would expect 2 things
- Invalid JSON should print out some error message, not crash the whole shell
- This is actually a valid [JSON Lines](http://jsonlines.org/) format. If I rename the file to `test.jsonl`, it works. However, I've seen JSON Lines being used with `.json` as well. Maybe non-standard, however it would be nice to support it...

**Configuration (please complete the following information):**
 - OS: Debian testing
 - Version 0.7.1

**Comment by Sosthene-Guedon** [comment]
Are you able to reproduce the issue with newer versions of Nushell? I'm not able to reproduce it on 0.11.0, however there is still a bug (see issue #1493)

**Comment by thegedge** [comment]
Yeah, same here as @Sosthene-Guedon. Doesn't crash, but also doesn't provide any useful error message. If that's the case with the latest version, @marekdedic, let's switch this to an enhancement report to report the invalid format.

> This is actually a valid JSON Lines format. If I rename the file to test.jsonl, it works. However, I've seen JSON Lines being used with .json as well. Maybe non-standard, however it would be nice to support it...

I've also seen this format called `njson` (newline-delimeted JSON). `from-json` has a `-o` flag that you can use right now:

```
cat <file> | from-json -o
```

I think I'd rather we either a) make it work with `open`, or b) make it so your reflex is to do something like `cat <file> | lines | from-json` instead of remembering Yet Another Flag™.

**Comment by marekdedic** [comment]
Hi, I can confirm no more crashes on 0.11.0 and also silent failures. I am for option a) - make it work with `open` as it is a fairly common data format :)

**Comment by Sosthene-Guedon** [comment]
The issue of silent failures has been fixed by PR  #1495.
